### PR TITLE
TDL-17511-Add missing test case, TDL-17522-Fix typos in the fields

### PR DESCRIPTION
**Pull request by umangagrawal-crest**
# Description of change
TDL-17511 Add missing tap tester tests

Added following below integration tests:

- Start Date Test
- Automatic Fields Test
- All Fields Test
- Bookmark Test
- Discovery Test
**NOTE:** Pagination test cases are not written as no stream is using pagination.

TDL-17522 FIx typos in the  fields

- Updated the field name `largest_purchase item_price` in the schema file of blast_save_list stream.
- Updated the streams.py file to use the correct replication_key.

**NOTE:** 

- Removed the test_sync_canary.py file.
- Added the config.yml file 
- The CircleCI plan is not configured for this tap.

# Manual QA steps
 - 
 
# Risks
 -  No risks
 
# Rollback steps
 - revert this branch

### CODE REVIEW

File: tap_sailthru/schemas/blast_save_list.json [hunk 1]
```diff
@@ -158,7 +158,7 @@
       ],
       "format": "date-time"
     },
-    "largest_purchase item_price": {
+    "largest_purchase_item_price": {
```
Comment (umangagrawal-crest) [inline tap_sailthru/schemas/blast_save_list.json:161]: The data was not collected for this field  `largest_purchase_item_price` due to a typo issue. So corrected the field name.

File: tap_sailthru/streams.py [hunk 2]
```diff
@@ -180,9 +180,9 @@ def sync(self,
         with metrics.record_counter(self.tap_stream_id) as counter:
             for record in self.get_records(bookmark_datetime):
                 self.date_records_to_datetime(record)
+                transform_keys_to_snake_case(record)
```
Comment (umangagrawal-crest) [inline tap_sailthru/streams.py:183]: Transforming record from camel_case to snake_case before retrieving a value of replication key from the record as replication_key is defined in snake_case in the  [schema file](https://github.com/singer-io/tap-sailthru/blob/main/tap_sailthru/schemas/purchase_log.json#L7)

Comment (umangagrawal-crest) [inline tap_sailthru/streams.py:183]: Transforming record from camel_case to snake_case before retrieving a value of replication key from the record as replication_key is defined in snake_case in the schema file
https://github.com/singer-io/tap-sailthru/blob/249d3ae35ec1e929152608a78abff3d1f86cb386/tap_sailthru/schemas/purchase_log.json#L6-L13

File: tap_sailthru/streams.py [hunk 3]
```diff
@@ -442,14 +442,14 @@ class PurchaseLog(IncrementalStream):
     """
     tap_stream_id = 'purchase_log'
     key_properties = ['extid']
-    replication_key = 'Date'
-    valid_replication_keys = ['Date']
+    replication_key = 'date'
+    valid_replication_keys = ['date']
```
Comment (umangagrawal-crest) [inline tap_sailthru/streams.py:446]: As mentioned above, syncing with the schema file.

**Review by savan-chovatiya — APPROVED** [review]

File: tap_sailthru/streams.py (hunk 2 above)
Comment (dbshah1212) [reply tap_sailthru/streams.py:183]: Can you please attach the card related to this bug? To explain this issue in detail.

Comment (umangagrawal-crest) [reply tap_sailthru/streams.py:183]: Here is the bug: https://jira.talendforge.org/browse/TDL-17522

**Review by dbshah1212 — APPROVED** [review]

**Review by karanpanchal-crest — APPROVED** [review]

**Review by kspeer825 — CHANGES_REQUESTED** [review]

File: tests/base.py [hunk 4]
```diff
@@ -1,6 +1,11 @@
 import os
 import unittest
-
+import singer
```
[the remark this answers is not in the document]
Comment (harshpatel4crest) [reply tests/base.py]: Removed.

File: tests/test_bookmark.py [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+        first_sync_record_count = self.run_and_verify_sync(conn_id)
+        first_sync_records = runner.get_records_from_target_output()
+        first_sync_bookmarks = menagerie.get_state(conn_id)
+
+        ##########################################################################
+        # Update State Between Syncs
+        ##########################################################################
+
+        new_states = first_sync_bookmarks
+        menagerie.set_state(conn_id, new_states)
```
[the remark this answers is not in the document]
Comment (harshpatel4crest) [reply tests/test_bookmark.py]: @kspeer825 Updated the state between the sync and set a date smaller then bookmark value.

**Review by kspeer825 — APPROVED** [review]

**Pull request merged**